eng/DownloadSharedSource.ps1

```diff
@@ -2,29 +2,89 @@
+    return $createResult
+}
+
+$sdkCloneDir = GetSDKCloneDir
+Write-Host "Setting up sparse clone for $gitRemoteUrl at $sdkCloneDir"
+
+Push-Location $sdkCloneDir.Path
+try {
+    if (!(Test-Path ".git")) {
+        Write-Host "Initializing sparse clone for $gitRemoteUrl"
+        InitializeSparseGitClone $gitRemoteUrl
+        AddSparseCheckoutPath $azCoreSharedPath
+        AddSparseCheckoutPath $armCoreSharedPath
+        $commitHash = git rev-parse $gitRemoteBranchName
+        git checkout $commitHash
+    }
+    if ($LASTEXITCODE) { exit $LASTEXITCODE }
+}
+finally {
+    Pop-Location
+}
+
+function CopyFilesToDestination([string]$sdkCloneRoot, [string]$dirName, [string]$dest, [string[]]$files) {
+    $source = Join-Path $sdkCloneRoot $dirName
+    foreach ($file in $files) {
+        $sourceFile = Join-Path $source $file
+        $destFile = Join-Path $dest $file
+        # to avoid the format issues (the file is required to end by a newline)
+        # here we read all the content in the source file, and then output it into the destination file
+        Write-Host "Copying $file"
+        $text = Get-Content -Path $sourceFile
+        $text | Out-File -FilePath $destFile
     }
 }
 
-$downloadPath = Resolve-Path (Join-Path $PSScriptRoot '..' 'src' 'assets' 'Azure.Core.Shared')
-Get-ChildItem $downloadPath -Filter *.cs | Remove-Item;
-$files = @('AsyncLockWithValue.cs', 'ClientDiagnostics.cs', 'DiagnosticScope.cs', 'DiagnosticScopeFactory.cs', 'ContentTypeUtilities.cs', 'HttpMessageSanitizer.cs',
-    'OperationInternalBase.cs', 'OperationInternal.cs', 'OperationInternalOfT.cs', 'TaskExtensions.cs', 'Argument.cs', 'Multipart/MultipartFormDataContent.cs',
-    'Multipart/MultipartContent.cs', 'AzureKeyCredentialPolicy.cs', 'AppContextSwitchHelper.cs',
-    'OperationPoller.cs', 'FixedDelayWithNoJitterStrategy.cs', 'SequentialDelayStrategy.cs',
-    'ForwardsClientCallsAttribute.cs', 'AsyncLockWithValue.cs', 'VoidValue.cs')
-$baseUrl = 'https://raw.githubusercontent.com/Azure/azure-sdk-for-net/main/sdk/core/Azure.Core/src/Shared/'
-DownloadAll $files $baseUrl $downloadPath
-
-#Download management Shared
-$files = 'SharedExtensions.cs', 'ManagedServiceIdentityTypeV3Converter.cs'
-$downloadPath = Resolve-Path (Join-Path $PSScriptRoot '..' 'src' 'assets' 'Management.Shared')
-Get-ChildItem $downloadPath -Filter *.cs | Remove-Item;
-$baseUrl = 'https://raw.githubusercontent.com/Azure/azure-sdk-for-net/main/sdk/resourcemanager/Azure.ResourceManager/src/Shared/'
-DownloadAll $files $baseUrl $downloadPath
+try {
+    Write-Host "Downloading Azure.Core.Shared"
+    $downloadPath = Resolve-Path (Join-Path $PSScriptRoot '..' 'src' 'assets' 'Azure.Core.Shared')
+    Write-Host "Removing the existing files in " $downloadPath
+    Get-ChildItem $downloadPath -Filter *.cs | Remove-Item;
+    $files = @('AsyncLockWithValue.cs', 'ClientDiagnostics.cs', 'DiagnosticScope.cs', 'DiagnosticScopeFactory.cs', 'ContentTypeUtilities.cs', 'HttpMessageSanitizer.cs',
```

Comment:
I this every file in the directory? Ideally we would sync the entire directory and not have to maintain this list of files.

Reply:
No. This is a subset we need. We only included things we need.